python/ray/train/accelerators/base.py

```diff
@@ -0,0 +1,2 @@
+class Accelerator:
```

Comment:
Okay to leave this name for now since this is completely internal, but wondering if we should rename (or split)  it in the future since we want it to have 2 responsibilities:

(1) "accelerate" optimizations
(2) "prepare" setup and capturing state (e.g. holding the model for checkpointing)

Reply:
I was also thinking that we could split the class up using a plugin design pattern to avoid violated the SRP